Ticket 91 View Flag Trends Page Question Buttons

bakolisty opened the pull request:
Added the following changes to the view flag trends page:

- Each question is now a button that will display a popup listing every audit that contains a red severity answer for that question.
- In the audit list popup the audit dates are buttons that will open up the completed audit that red severity question is in.

I needed to make small changes to the completed audit page, and completed audit list page to make the back button on the completed audit page dynamic. 

- Added a previous_page variable to the completed audit page class.
- Updated the on_press logic for the back button on the completed audit page to use the previous_page variable.
- Modified the callback method in the completed audit list page to set the previous page variable on the completed audit page.

This branch is already merged with ticket 145 so we should get that in master before this branch.

BaderAlshaya commented:
Noticed a couple of layout bugs. The first separating line are extending outside the scope of the Popup. Also, the first button seems like it's taking some space of the header row. 

Here is a screenshot with the line highlighted in red:
![Screen Shot 2019-12-02 at 7 34 11 PM](https://user-images.githubusercontent.com/18043111/70018679-c32d4880-153b-11ea-8be9-bc4e357a3db5.png)

BaderAlshaya commented:
I know this is not part of this ticket, but it's something I noticed while reviewing this ticket and it's worth mentioning (I thinks there is a bug that needs to be fixed asap). The number of flagged questions in the view_flagged_trends_page doesn't match the number of actual flagged answers provided in the new Popup. I think the `view_flagged_tends_page` needs to be reviewed for a potential logic error.

bakolisty commented:
> I know this is not part of this ticket, but it's something I noticed while reviewing this ticket and it's worth mentioning (I thinks there is a bug that needs to be fixed asap). The number of flagged questions in the view_flagged_trends_page doesn't match the number of actual flagged answers provided in the new Popup. I think the view_flagged_tends_page needs to be reviewed for a potential logic error.

Nice catch, I will look into that now.

bakolisty commented:
I fixed the issues mentioned last night so this PR should be good for more review. I had to remove pagination on the view flag trends page as that was what was causing the flagged answer count to be off. This change will cause the startup time of the application to increase but I can't think of a way where we can keep the pagination and have an accurate flagged answer count. I also changed the labels and buttons used for the popup which should fix the layout issues brought up. While testing these issues I noticed duplicate completed audits appearing in the popup like this:

![image](https://user-images.githubusercontent.com/11793240/70051182-68204380-1585-11ea-853f-ff60cc57569e.png)

This is caused by having the exact same question multiple times in the same audit template like seen below:

![image](https://user-images.githubusercontent.com/11793240/70051289-9e5dc300-1585-11ea-85af-bd0e4c3070f4.png)

This should not be an issue when the audit templates are created by someone though as I do not see much of a point in having duplicate questions in the same template.

BaderAlshaya commented:
It looks better now! I just pushed a new commit to replace Kivy's buttons with cilantro templates to make things consistent, and I've done minor formatting.
In regard to the duplicate questions you noticed, I noticed the same things a few days earlier and I already added a fix as mentioned in this PR #159 (great catch tho).
